crml/cennzx/src/lib.rs

```diff
@@ -781,29 +787,29 @@ impl<T: Trait> Module<T> {
 			} else {
 				T::ExchangeAddressFor::exchange_address_for(asset_to_buy)
 			};
-			let _ = T::AssetSystem::transfer(asset_to_sell, trader, &exchange_address, amount_to_sell).and(
+
+			T::AssetSystem::transfer(asset_to_sell, trader, &exchange_address, amount_to_sell).and(
```

Comment:
should **.and(** be like
**and_then**  like in 
Self::get_asset_to_core_sell_price(asset_to_sell, amount_to_sell).and_then(|core_amount| {
 				T::AssetSystem::transfer(asset_to_sell, trader, &exchange_address_a, amount_to_sell)

Reply:
afaik `and` will return err if either error
in the 2nd case I used `and_then` so I could pass through the `core_amount` from price calculation